src/components/FlightsTable/FlightsTable.jsx

```diff
@@ -7,9 +7,9 @@ export const FlightsTable = ({originFilter, flights}) => {
     <section className="flightsTable">
       <h3>Flights</h3>
       <section>
-        {flights.filter((flight) => (
+        {flights?.filter((flight) => (
           originFilter.length ? originFilter.includes(flight.origin) : flight
-        )).map((flight) => (
+        ))?.map((flight) => (
```

Comment:
curious about this...
do you need to check the length of the originFilter array? if the length exists, you're then checking if flight includes .origin object... 
what happens if you just call .map() on the flights array directly? why bother checking the length and .origin?

Reply:
Probably an indication that I could write this better. But let me tell you my intent and then perhaps you can help me figure out a better path.

The filter section is saying, if there is an originFilter being applied, then filter to ONLY include flights where the origin matches the entry(ies) in the originFilter. If there is no originFilter being applied, then pass in all the flights.

So if a flights origin is BOS but the originFilter is ["DEN"], it would not be passed into the map function. And if there was an error fetching flights, the ? is to prevent an error from being thrown on the mapping function.

Does that make sense? How can I do this more efficiently or in a way thats easier to read?